### DISCUSSION TITLE
Assist with rtxmlrpc usage

### DISCUSSION THREAD

**Discussion by govardha** [kannibalox/pyrosimple#70] [General]
Hello kannibalox,

Cam you give me an assist with the rtxmlrpc usage.  I am trying to use the has of a torrent file and execute the https://rtorrent-docs.readthedocs.io/en/latest/cmd-ref.html#term-event-download-paused event or something else.  I am using this to execute some bash_handler scripts.

Thank you!
G

**Comment by kannibalox** [comment]
You can simply call the hash with the event name as the command, which will then run all the linked commands:
```bash
rtxmlrpc event.download.paused 81AF07491915415DAD45F87C0C2AE52FAE92C06B
```

You can also run it against multiple items with rtcontrol:
```
rtcontrol // --exec event.download.paused=
```

**Comment by govardha** [comment]
Hello @kannibalox 

Thanks again! for your help.  I am learning a lot from reading your code and I am attempting to debug why the rtcontrol with the cull or purge switches don’t remove both the metadata + data, it just removes the metadata file right now. 

I am assuming rtcontrol under the hood is making a d.multicall2.  Can you share some of your debug tricks when you were starting to work on porting over the python2 version into python3. 

Thank you!

**Comment by kannibalox** [comment]
> I am attempting to debug why the rtcontrol with the cull or purge switches don’t remove both the metadata + data, it just removes the metadata file right now.

I'll respond to this in the other discussion.

> I am assuming rtcontrol under the hood is making a d.multicall2. Can you share some of your debug tricks when you were starting to work on porting over the python2 version into python3.

It does an initial d.multicall2 to get the torrent information, but then action flags are execute one at a time. Debug logging (`--debug`) on any of the tools will show you all the RPC commands it's executing. The conversion from python2 to python3 was mostly refactoring, rather than troubleshooting rTorrent directly. I used a lot of linting and unit testing to ensure functionality didn't change.
